src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -328,7 +329,9 @@ private string GenerateId()
                 //sanitize external RequestId as it may not be hierarchical. 
                 //we cannot update ParentId, we must let it be logged exactly as it was passed.
                 string parentId = ParentId[0] == RootIdPrefix ? ParentId : RootIdPrefix + ParentId;
-                if (parentId[parentId.Length - 1] != '.')
+
+                char lastChar = parentId[parentId.Length - 1];
+                if (lastChar != '.' && lastChar != '_')
```

Comment:
> so I don't think it worth testing

But some corner cases would break if we ever deleted this by accident? Testing costs us very little; why not have one?